packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1035,12 +1037,23 @@ class RenderParagraph extends RenderBox
   void describeSemanticsConfiguration(SemanticsConfiguration config) {
     super.describeSemanticsConfiguration(config);
     _semanticsInfo = text.getSemanticsInformation();
+    bool needsAssembleSemanticsNode = false;
+    bool needsChildConfigrationsDelegate = false;
+    for (final InlineSpanSemanticsInformation info in _semanticsInfo!) {
+      if (info.recognizer != null) {
+        needsAssembleSemanticsNode = true;
+        break;
+      }
+      needsChildConfigrationsDelegate = needsChildConfigrationsDelegate || info.isPlaceholder;
+    }
 
-    if (_semanticsInfo!.any((InlineSpanSemanticsInformation info) => info.recognizer != null)) {
+    if (needsAssembleSemanticsNode) {
```

Comment:
It's not clear to me why we can skip the delegate if there's a recognizer. Don't we still have the same problem with the placeholder nodes in this case?

Reply:
If there is gesture recognizer, every continuous strings are separated into its own nodes so there won't be merging issue. and the assemblesemanticsnode is smart enough to insert each semanticsnode of widgetspans in the right places. So there is no need to use the childconfigurationdelegate API. As we discussed offline, if we do want to replace assemblesmeanticsnode with the childconfigurationdelegate API we need a way to set stable id and geometry

Reply:
Makes sense. Thanks for the explanation.